### DISCUSSION TITLE
Welcome to wis2box Discussions!

### DISCUSSION THREAD

**Discussion by tomkralidis** [wmo-im/wis2box#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by maixino** [comment]
I am trying to run  python3 wis2box-ctl.py start .
Here is the output
ulling grafana (grafana/grafana-oss:latest)...
Trying to pull repository docker.io/grafana/grafana-oss ... 
latest: Pulling from docker.io/grafana/grafana-oss
df9b9388f04a: Pull complete
28375ec17b8f: Pull complete
ca6ee70dc807: Pull complete
ecf4706c932b: Pull complete
b35722c54709: Pull complete
aa885cc7e6df: Pull complete
1a9d41482cce: Pull complete
f29b04feb70b: Pull complete
d210684d2126: Pull complete
Digest: sha256:b0dcebece46a323daabe07da2bd1df2ae0ba5e4cfe2e6331ab234181ab159a62
Status: Downloaded newer image for docker.io/grafana/grafana-oss:latest
Building metrics_collector
unknown flag: --iidfile
See 'docker build --help'.
ERROR: Service 'metrics_collector' failed to build : Build failed

**Comment by maixino** [comment]
After re-installing, I am trying to run python3 wis2box-ctl.py start  again
Here is the output
Traceback (most recent call last):
  File "urllib3/connectionpool.py", line 677, in urlopen
  File "urllib3/connectionpool.py", line 392, in _make_request
  File "http/client.py", line 1277, in request
  File "http/client.py", line 1323, in _send_request
  File "http/client.py", line 1272, in endheaders
  File "http/client.py", line 1032, in _send_output
  File "http/client.py", line 972, in send
  File "docker/transport/unixconn.py", line 43, in connect
PermissionError: [Errno 13] Permission denied

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "requests/adapters.py", line 449, in send
  File "urllib3/connectionpool.py", line 727, in urlopen
  File "urllib3/util/retry.py", line 410, in increment
  File "urllib3/packages/six.py", line 734, in reraise
  File "urllib3/connectionpool.py", line 677, in urlopen
  File "urllib3/connectionpool.py", line 392, in _make_request
  File "http/client.py", line 1277, in request
  File "http/client.py", line 1323, in _send_request
  File "http/client.py", line 1272, in endheaders
  File "http/client.py", line 1032, in _send_output
  File "http/client.py", line 972, in send
  File "docker/transport/unixconn.py", line 43, in connect
urllib3.exceptions.ProtocolError: ('Connection aborted.', PermissionError(13, 'Permission denied'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker/api/client.py", line 214, in _retrieve_server_version
  File "docker/api/daemon.py", line 181, in version
  File "docker/utils/decorators.py", line 46, in inner
  File "docker/api/client.py", line 237, in _get
  File "requests/sessions.py", line 543, in get
  File "requests/sessions.py", line 530, in request
  File "requests/sessions.py", line 643, in send
  File "requests/adapters.py", line 498, in send
requests.exceptions.ConnectionError: ('Connection aborted.', PermissionError(13, 'Permission denied'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 81, in main
  File "compose/cli/main.py", line 200, in perform_command
  File "compose/cli/command.py", line 70, in project_from_options
  File "compose/cli/command.py", line 153, in get_project
  File "compose/cli/docker_client.py", line 43, in get_client
  File "compose/cli/docker_client.py", line 170, in docker_client
  File "docker/api/client.py", line 197, in __init__
  File "docker/api/client.py", line 222, in _retrieve_server_version
docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', PermissionError(13, 'Permission denied'))
[10492] Failed to execute script docker-compose

**Comment by maaikelimper** [comment]
Hi [Maixentkambi](https://github.com/Maixentkambi),

Thank you for trying out wis2box !

Can you confirm docker is running using the following command:
`systemctl status docker.service`

Could you let us know the version of docker and docker-compose you are running ?
`docker --version`
`docker-compose --version`

**Reply by maixino** [reply]
Hi maaikelimper

thanks for your answer . Here is the output

**root@localhost wis2box]# systemctl status docker.service**
● docker.service - Docker Application Container Engine
   Loaded: loaded (/usr/lib/systemd/system/docker.service; disabled; vendor preset: disabled)
   Active: active (running) since Sat 2022-04-30 13:49:59 WAT; 2 days ago
     Docs: http://docs.docker.com/
Main PID: 10065 (dockerd-current)
    Tasks: 48
   CGroup: /[system.slice/docker.service](http://system.slice/docker.service)
           ├─10065 /usr/bin/dockerd-current --add-runtime docker-runc=/usr/libexec/docker/docker-runc-current --default-runtime=docker-runc --exec-opt native.cgroupdriver=systemd --userland-proxy...
           └─10077 /usr/bin/docker-containerd-current -l unix:///var/run/docker/libcontainerd/docker-containerd.sock --metrics-interval=0 --start-timeout 2m --state-dir /var/run/docker/libcontain...

Apr 30 14:35:00 localhost.localdomain dockerd-current[10065]: time="2022-04-30T14:35:00.208846600+01:00" level=error msg="Error setting up exec command in container wis2box: No such co...r: wis2box"
Apr 30 14:35:00 localhost.localdomain dockerd-current[10065]: time="2022-04-30T14:35:00.208880059+01:00" level=error msg="Handler for POST /v1.26/containers/wis2box/exec returned error...r: wis2box"
Apr 30 15:21:07 localhost.localdomain dockerd-current[10065]: time="2022-04-30T15:21:07.092878627+01:00" level=error msg="Handler for GET /v1.26/networks/wis2boxproject_default returne... not found"
Apr 30 15:21:07 localhost.localdomain dockerd-current[10065]: time="2022-04-30T15:21:07.095892129+01:00" level=error msg="Handler for GET /v1.26/volumes/wis2boxproject_es-data returned...uch volume"
Apr 30 15:21:07 localhost.localdomain dockerd-current[10065]: time="2022-04-30T15:21:07.098411009+01:00" level=error msg="Handler for GET /v1.26/volumes/wis2boxproject_prometheus-data ...uch volume"
Apr 30 15:21:07 localhost.localdomain dockerd-current[10065]: time="2022-04-30T15:21:07.101272745+01:00" level=error msg="Handler for GET /v1.26/networks/wis2boxproject_vpcbr returned ... not found"
Apr 30 15:21:07 localhost.localdomain dockerd-current[10065]: time="2022-04-30T15:21:07.103628938+01:00" level=error msg="Handler for GET /v1.26/volumes/wis2boxproject_loki-data return...uch volume"
Apr 30 15:21:07 localhost.localdomain dockerd-current[10065]: time="2022-04-30T15:21:07.212829640+01:00" level=error msg="Handler for GET /v1.26/images/prom/prometheus/json returned er...prometheus"
Apr 30 15:44:44 localhost.localdomain dockerd-current[10065]: time="2022-04-30T15:44:44.113177094+01:00" level=error msg="Download failed, retrying: read tcp 192.168.2.61:59594->104.18...et by peer"
Apr 30 15:46:54 localhost.localdomain dockerd-current[10065]: time="2022-04-30T15:46:54.183763089+01:00" level=error msg="Handler for GET /v1.26/images/wis2box_project_metrics_collecto..._collector"
Hint: Some lines were ellipsized, use -l to show in full.
**[root@localhost wis2box]# docker --version**
Docker version 1.13.1, build 7d71120/1.13.1
**[root@localhost wis2box]# docker-compose --version**
docker-compose version 1.29.2, build 5becea4c
[root@localhost wis2box]#

**Reply by maaikelimper** [reply]
Thanks. The docker version looks to be old. Could you please install the latest version of docker: https://docs.docker.com/engine/install/

My output for docker --version
`Docker version 20.10.12, build e91ed57`

**Comment by maixino** [comment]
Hi all

thanks for your answer . Here is the output

[root@localhost wis2box]# python3 wis2box-ctl.py start
Trying to pull repository docker.io/grafana/loki-docker-driver ... 
latest: Pulling from docker.io/grafana/loki-docker-driver
32dbe5a742d5: Downloading [==================================================>] 20.42 MB/20.42 MB
failed to download: unexpected EOF
Creating network "wis2box_project_default" with the default driver
Creating network "wis2box_project_vpcbr" with driver "bridge"
Building mosquitto
unknown flag: --iidfile
**See 'docker build --help'.
ERROR: Service 'mosquitto' failed to build : Build failed**

**Reply by gespau** [reply]
I think this was addressed by #148

**Comment by sturmanjd** [comment]
Hello,
I need to change the ports for a test installation of WIS2BOX.
Is it just the docker-compose.override.yml file I need to adjust and enter the new port numbers or will I also need to modify the docker-compose.yml and default.env file?
Thanks

**Reply by sturmanjd** [reply]
As an update I have modified the port numbers in the docker-compose.override.yml file and modified the dev.env file to use the new ports to set the WIS2BOX_API_URL and WIS2BOX_URL

I haven't changed the docker-compose.yml file yet.

I followed the quickstart instructions including the ingest data and cache/publish stations steps to load the tests data. I can see WIS2BOX running in a browser, but I can't see any data. Should I be able to see data on these pages?

![image](https://user-images.githubusercontent.com/119272382/204432131-903b8f67-ce91-462e-8f0f-43d51c21f985.png)

![image](https://user-images.githubusercontent.com/119272382/204432270-01dfcc50-7b30-4176-a257-d1d412661025.png)

Any assistance would be gratefully received.

**Comment by maaikelimper** [comment]
Hi,

By default the WIS2BOX_URL=http://localhost:8999 
When you update the ports used in docker-composer.override.yml you will also need to redefine the WIS2BOX_URL and WIS2BOX_API_URL environment-variable.

It sounds like that's exactly what you did, but to double-check, could you repeat these steps:

For example,
change the port binding on localhost from port 8999 to port 1234 in docker-compose.override.yml
```
  web-proxy:
    ports:
      - 1234:80
```

Update WIS2BOX_URL and WIS2BOX_API_URL in dev.env :
```
WIS2BOX_URL=http://localhost:1234
WIS2BOX_API_URL=http://localhost:1234/oapi
```

Restart the wis2box-stack
```
python wis2box-ctl.py stop
python wis2box-ctl.py start
```
And WIS2BOX_UI is available on http://localhost:1234

If your wis2box shows no datasets, it might mean you have not added the collections. 

To add collections, login to the wis2box-container:
```
python wis2box-ctl.py login
````

and, assuming your using the test-data from the quickstart, run the following commands:

add-collection
```
wis2box data add-collection $WIS2BOX_DATADIR/metadata/discovery/mwi-surface-weather-observations.yml
wis2box data add-collection $WIS2BOX_DATADIR/metadata/discovery/ita-surface-weather-observations.yml
wis2box data add-collection $WIS2BOX_DATADIR/metadata/discovery/dza-surface-weather-observations.yml
```
publish metadata
```
wis2box metadata discovery publish $WIS2BOX_DATADIR/metadata/discovery/mwi-surface-weather-observations.yml
wis2box metadata discovery publish $WIS2BOX_DATADIR/metadata/discovery/ita-surface-weather-observations.yml
wis2box metadata discovery publish $WIS2BOX_DATADIR/metadata/discovery/dza-surface-weather-observations.yml
````

**Comment by maaikelimper** [comment]
We are working on improving documentation for WIS2-in-a-box users with a new guide available here:
https://wis2box-user-guide.readthedocs.io

**Reply by gespau** [reply]
The current docs are available at: https://docs.wis2box.wis.wmo.int/

**Comment by willself** [comment]
Hi

I've been working to get wis2box running but once I'm set up and applied some preliminary config from the guide, the containers don't seem to deploy. The output is listed below.

When setting up the host, I had to remove docker-compose-plugin (yum remove ...) as the version was > v2 and get docker-compose 1.29.2 manually and then ln -s /usr/bin/docker-compose according to the troubleshooting guide.

I've tested docker-compose using the example at https://docs.docker.com/compose/gettingstarted/ and that example worked normally with docker, so I think docker and docker-compose are working fine. 

docker version is 23.0.5

After attempting to start wis2box (output below) "docker info" reports no containers running but I'm left with three docker-proxy processes for ports 8884. 1883 and 3000. Those persist across docker stop and starts and I have to remove them from /var/lib/docker/network/files/local-kv.db, otherwise they interfere with following wis2box start attempts. 

I've switched SELinux to permissive and stopped firewalld. I tried removing all the containers and images with docker rm(i) but once everything was retrieved again the same as below occurred. I tried various entries for DOCKER_HOST env but that didn't seem to help.

Do you have any tips or can see what I've got wrong here? 

```
python3 wis2box-ctl.py start
Error response from daemon: plugin loki:latest already exists
Error response from daemon: plugin already enabled: plugin loki:latest is enabled
Creating network "wis2box_project_default" with the default driver
Creating network "wis2box_project_vpcbr" with driver "bridge"
Creating mosquitto ...
Creating grafana   ...
Creating prometheus ...
Creating loki       ...
Creating elasticsearch ...

ERROR: for grafana  ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

ERROR: for elasticsearch  ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

ERROR: for prometheus  ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

ERROR: for loki  ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

ERROR: for mosquitto  ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

ERROR: for grafana  ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

ERROR: for elasticsearch  ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

ERROR: for prometheus  ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

ERROR: for loki  ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

ERROR: for mosquitto  ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
ERROR: Couldn't connect to Docker daemon at http+docker://localhost - is it running?

If it's at a non-standard location, specify the URL with the DOCKER_HOST environment variable.
```

Thank you for any help you can give
Bill Self

**Reply by maaikelimper** [reply]
From what you described the problem seems related to your Docker-installation. Here are a few things you could try.

You can test Docker works OK by running the hello-world example:

`docker run -it hello-world`

Clean up your Docker system before attempting to start wis2box, you can use `docker system df` to check if you need to remove any previous container/volumes. 

You could try to start without the monitoring-stack (no cAdvisor/grafana/loki/prometheus):

`docker-compose --file docker-compose.yml --file docker-compose.override.yml --env-file dev.env --project-name wis2box_project up -d`

**Reply by willself** [reply]
Hi

Thank you for getting back so quickly. The hello world example worked fine and I had tried clearing docker but unfortunately didn't help.

Starting without the monitoring as you describe though has got me further along with the following

```
docker-compose --file docker-compose.yml --file docker-compose.override.yml --env-file dev.env --project-name wis2box_project up -d
Creating network "wis2box_project_default" with the default driver
Creating mosquitto     ... done
Creating elasticsearch ... done
Creating wis2box-minio ... done
Creating wis2box-api   ... done
Creating wis2box-ui         ... done
Creating wis2box-management ... done
Creating nginx              ... done
Creating wis2box-auth       ... done

python3 wis2box-ctl.py status
       Name                     Command                  State                           Ports
-------------------------------------------------------------------------------------------------------------------
elasticsearch        /bin/tini -- /usr/local/bi ...   Up (healthy)   9200/tcp, 9300/tcp
mosquitto            /docker-entrypoint.sh /usr ...   Up             0.0.0.0:1883->1883/tcp, 0.0.0.0:8884->8884/tcp
nginx                /docker-entrypoint.sh ngin ...   Restarting
wis2box-api          /app/docker/es-entrypoint.sh     Up
wis2box-auth         /entrypoint.sh                   Up
wis2box-management   /entrypoint.sh wis2box pub ...   Up
wis2box-minio        /usr/bin/docker-entrypoint ...   Up (healthy)   0.0.0.0:9000->9000/tcp, 0.0.0.0:9001->9001/tcp
wis2box-ui           /docker-entrypoint.sh ngin ...   Exit 1
```

and I see containers running, (where as before there was never anything listed despite starting wis2box-ctl.py)
```
docker ps
CONTAINER ID   IMAGE                                                  COMMAND                  CREATED         STATUS                         PORTS                                            NAMES
2017bf5a432c   nginx:alpine                                           "/docker-entrypoint.…"   2 minutes ago   Restarting (1) 9 seconds ago                                                    nginx
533fc98d00c4   ghcr.io/wmo-im/wis2box-auth:0.5.0                      "/entrypoint.sh"         2 minutes ago   Up 2 minutes                                                                    wis2box-auth
ece9c9697de9   ghcr.io/wmo-im/wis2box-management:1.0b2-1              "/entrypoint.sh wis2…"   2 minutes ago   Up 26 seconds                                                                   wis2box-management
983314dd57e2   minio/minio:RELEASE.2022-12-02T19-19-22Z.fips          "/usr/bin/docker-ent…"   2 minutes ago   Up 2 minutes (healthy)         0.0.0.0:9000-9001->9000-9001/tcp                 wis2box-minio
fadd561b13da   ghcr.io/wmo-im/wis2box-broker:1.0b2                    "/docker-entrypoint.…"   2 minutes ago   Up 2 minutes                   0.0.0.0:1883->1883/tcp, 0.0.0.0:8884->8884/tcp   mosquitto
1e6954148e9e   docker.elastic.co/elasticsearch/elasticsearch:7.17.0   "/bin/tini -- /usr/l…"   2 minutes ago   Up 2 minutes (healthy)         9200/tcp, 9300/tcp                               elasticsearch

```
http://localhost/oapi/collections doesn't return anything yet (not listening on port 80) but may be that's expected starting things like this?

Does that reveal anything about what might be wrong with my installation and what I can try from here? Stopping everything, clearing docker and trying to start with "python3 wis2box-ctl.py start" still results in the problems listed earlier.

I'm not sure the started services are stable though; nginx is always restarting and wis2box-api and ui report Exit 1 as their status. Running the same command with 'down' instead of 'up' stops everything cleanly.

Thank you again for your help, very appreciated.
Bill

**Reply by maaikelimper** [reply]
From the information your provided it looks like the nginx-container is restarting. You could check the logs for this container with `docker logs nginx` to get an indication what might be going wrong. Please also check if you already have a service running on your instance that uses port 80 as that would prevent the nginx-container from starting.

**Reply by willself** [reply]
Hi

Sorry for late reply. We found that ipv6 was shut down on the host and some other features were disabled for instance hardening which stopped wis2bix running normally. Creating a standard ec2 instance and wis2box ran first time and without issues.

We have our test instance running now for 2 or 3 months.

Thank you again for your help earlier.
Bill

**Comment by jandegithub** [comment]
Typing to get v1.0b6 running (same problem with git-main) 
Probably indicates a problem with my machine ?

Cleared out all the docker container.
Followed instructions to install at [wmo setup](https://docs.wis2box.wis.wmo.int/en/1.0b6/user/setup.html)

> python3 wis2box-create-config.py
Want to test locally so URL is http://localhost

> python3 wis2box-ctl.py start

``` 
⠿ Container elasticsearch           Waiting                                                                                        66.0s
container for service "elasticsearch" is unhealthy
```

docker inspect --format "{{json .State.Health }}" elasticsearch | jq
```    
{
      "Start": "2024-02-06T13:16:16.919727503Z",
      "End": "2024-02-06T13:16:17.093942703Z",
      "ExitCode": 7,
      "Output": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (7) Failed to connect to localhost port 9200: Connection refused\n"
    }
```

So it seems that elasticsearch is not accepting connections. I have no idea why and need help to get this running
